### PR TITLE
Add representative selection process guide.

### LINKED ISSUE #25: Consider creating suggestions on processes for teams to pick a Council representative
Currently there isn't any suggestion for how teams should pick their representative other than "figure it out on your own", which can lead to challenges for teams to pick a process that is efficient and produces good results. I would suggest considering writing down a suggested process for top-level teams to refer to as a guide for having a process that we think will be smooth and effective. But I think we don't necessarily need to *require* any particular process to be used (though I don't know).

Similar to this is suggestions on how to deal with alternates. Is it useful for teams to have a standing alternate? Or is it best to pick alternates on an as-needed basis? Should you use the same process for picking an alternate as a regular member.

Similarly, how can the selection process be aware of bias in Council representative selection, including status-quo bias towards existing Rust leaders or people similar to them?

### PR DESCRIPTION
**Pull request by ehuss**
This adds a guide to suggest how teams may carry out their council representative selection process. This is largely taken from https://github.com/rust-lang/rfcs/pull/3392#issuecomment-1505697944, with various small edits.

Closes #25

### CODE REVIEW

**Review by eholk — APPROVED** [review]

File: guides/representative-selection.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Representative selection process
+
+This document is a guide for the selection of Representatives for the Leadership Council. This gives recommendations for how a top-level team should select their Representative. It is not a requirement that teams follow this guide, and may choose their own process.
```
Comment (jonathanpallant) [inline guides/representative-selection.md:3]:
```suggestion
This document is a guide for the selection of Representatives for the Leadership Council. This gives recommendations for how a top-level team should select their Representative. It is not a requirement that teams follow this guide - teams may choose their own process.
```

File: guides/representative-selection.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+This process is inspired by the [Sociacracy selection process](https://www.sociocracyforall.org/selection-process/).
```
Comment (jonathanpallant) [inline guides/representative-selection.md:5]:
```suggestion
This process is inspired by the [Sociocracy selection process](https://www.sociocracyforall.org/selection-process/).
```

File: guides/representative-selection.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+## Recommendations on candidate choice
+
+See [candidate criteria] for guidance for choosing a candidate. See [representative role] for a description of what a Representative is expected to do.
+
+Many of the same qualities that make a good Council representative also make a good team lead. However, we strongly recommend against individuals taking on both the role of team lead and Council representative simultaneously. In the past, our experience has shown that it is extremely difficult to balance the demands of both roles effectively. For teams with shared leadership responsibilities (e.g. co-leads), assigning one co-lead as the representative may be fine if the other co-lead(s) are prepared to take on more of the team leadership responsibilities. However, for teams with a single lead, we recommend either establishing multiple clearly delineated team leadership roles or selecting a different Council representative.
+
+Remember that Council representatives can be subteam members, if a subteam member has the requisite scope and skillset.
```
Comment (jonathanpallant) [inline guides/representative-selection.md:13]:
```suggestion
Remember that Council representatives can be subteam members, if a subteam member has the requisite scope and skillset. For the avoidance of doubt, working groups count as subteams for this purpose.
```

File: guides/representative-selection.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+6. Finalize representative selection:
+    - Discuss and attempt to resolve any objections, if possible.
+    - If the proposed candidate has no remaining objections, select them as the Council representative.
+    - If objections persist and the current proposal reaches an impasse, the facilitator should make a new proposal based on the discussion and feedback received.
+    - Continue the process of seeking consensus and addressing objections until a satisfactory candidate is selected.
+        - Note that this process seeks to produce a candidate everyone is satisfied with and has no objections to. Avoid lengthy debates over preferences that do not pertain to objections or unmet requirements.
+7. Report the choice to the Leadership Council.
+    - The Council is responsible for validating that the new Council composition does not exceed the [affiliation limits]. If there is an exceedance, see the given link for the process of restarting the selection process.
```
Comment (jonathanpallant) [inline guides/representative-selection.md:47]: Which 'given link'? `[affiliation limits]`?

Comment (ehuss) [reply guides/representative-selection.md]: Yes, `[affiliation limits]`.

File: guides/representative-selection.md (hunk 3 above)
Comment (ehuss) [reply guides/representative-selection.md]: I added a complete definition of what is meant here.

File: guides/representative-selection.md [hunk 5]
```diff
@@ -1,16 +1,16 @@
 # Representative selection process
 
-This document is a guide for the selection of Representatives for the Leadership Council. This gives recommendations for how a top-level team should select their Representative. It is not a requirement that teams follow this guide, and may choose their own process.
+This document is a guide for the selection of Representatives for the Leadership Council. This gives recommendations for how a top-level team should select their Representative. It is not a requirement that teams follow this guide; teams may choose their own process.
```
Comment (jonathanpallant) [inline guides/representative-selection.md:3]: Given the clarification added, perhaps this should now be:

> ; top-level teams may choose their own process.

I wouldn't want to suggest that any team many choose their own process, in conflict with their parent top-level team.

**Comment by ehuss** [comment]
@rust-lang/leadership-council Just wanted to confirm everyone agrees with this proposal. It is not a required policy, just a suggested guide for teams to follow.

@rfcbot fcp merge

**Pull request merged**